orb.yml

```diff
@@ -106,12 +110,39 @@ commands:
             elif uname -a | grep Darwin > /dev/null 2>&1; then
               export PATH=$PATH:/tmp/buildevents/be/bin-darwin:/tmp/be/bin-darwin
             fi
+
+            # if there are any extra context values, add them
+            if [ -e /tmp/buildevents/extra_fields.lgfmt ] ; then
+              export BUILDEVENT_FILE=/tmp/buildevents/extra_fields.lgfmt
+            fi
+
+            # go ahead and report the span
             buildevents step $CIRCLE_WORKFLOW_ID \
               $(cat /tmp/buildevents/be/${CIRCLE_JOB}/span_id) \
               $(cat /tmp/buildevents/be/${CIRCLE_JOB}/start) \
               $CIRCLE_JOB
           when: always
 
+  add_context:
+    description: |
+      add_context is a function to add additional fields to the span
+      representing this job. It can be called multiple times, each with one
+      value. It is useful for adding things like artifact sizes, parameters to
+      the job, and so on. Call this with two arguments - the field name and the
+      field value. Both name and value must be strings, but numbers will be
+      coerced before getting sent to Honeycomb. Names must be single words;
+      values can be longer strings. This must be used within the context of
+      `with_job_span`
+    parameters:
+      field_name:
+        description: the name of the field to add to the surrounding step
+        type: string
+      field_value:
+        description: the value of the field to add to the surrounding step
+        type: string
+    steps:
+      - run: echo '<< parameters.field_name >>="<< parameters.field_value >>"' >> /tmp/buildevents/extra_fields.lgfmt
```

Comment:
I don't suppose we can leave the quoting up to the user somehow?  otherwise this can break if they use quotes..

Reply:
the circle parameters appear to swallow quotes. at least some of the time.